s3access/s3access.py

```diff
@@ -48,10 +46,13 @@ def read_value(value, type):
         return None
 
 
+T = TypeVar('T')
+
+
 class S3Access:
-    def __init__(self, parallelism: int = multiprocessing.cpu_count() * 4):
-        self._num_workers = parallelism
-        self._cachedir = os.getenv('S3ACCESSCACHE')
+    def __init__(self, parallelism: int = multiprocessing.cpu_count() * 4, cachedir: str = os.getenv('S3ACCESSCACHE')):
```

Comment:
setting the 'cachedir' like that will create problems if importing before setting the environment variable (good for services bad for notebooks)

Reply:
fixed